pkg/asset/templates/kubelet.yaml

```diff
@@ -17,27 +17,19 @@ spec:
       - name: kubelet
         image: quay.io/coreos/hyperkube:v1.3.0_coreos.1
         command:
-        - /nsenter
-        - --target=1
-        - --mount
-        - --wd=.
-        - --
-        - ./hyperkube
-        - kubelet
-        - --api-servers={{ index .APIServers 0 }}
-        - --config=/etc/kubernetes/manifests
-        - --allow-privileged
-        - --hostname-override=$(MY_POD_IP)
-        - --cluster-dns=10.3.0.10
-        - --cluster-domain=cluster.local
-        - --kubeconfig=/etc/kubernetes/kubeconfig
-        - --lock-file=/var/run/lock/kubelet.lock
-        - --minimum-container-ttl-duration=3m0s
+        - /bin/bash
+        - -c
+        - "/nsenter --target=1 --mount --wd=. -- ./hyperkube kubelet --hostname-override=$(MY_POD_IP) $(FLAGS)"
         env:
           - name: MY_POD_IP
             valueFrom:
               fieldRef:
                 fieldPath: status.podIP
+          - name: FLAGS
+            valueFrom:
+              configMapKeyRef:
+                name: kubelet-config
```

Comment:
We could merge those flags if we go with the option of using the on-disk config for both on-host and self-hosted. For example, we could make it so that the node-agent will look for the config named in the annotation and then `$configName-$nodeName` as well, and merge those flags. That way we keep a single daemonset, but get the flexibility of node specific config. This is mimicing what the proposed upstream kubelet config would be doing too, correct?